nodejs/eks/cluster.ts

```diff
@@ -579,9 +596,68 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
         nodeSecurityGroupTags: args.nodeSecurityGroupTags,
         storageClasses: userStorageClasses,
         fargateProfile: fargateProfile,
+        oidcProvider: oidcProvider,
     };
 }
 
+// Adapted from https://git.io/JvGHB
+
+// Find the root CA cert in a chain of certs.
+function findCACertificate(certificate: any, parent?: any): any {
+    if (certificate?.issuerCertificate.fingerprint !== certificate.fingerprint) {
+      return findCACertificate(certificate.issuerCertificate, certificate);
+    }
+    return parent;
+}
+
+const THUMBPRINT_MAX_RETRIES: number = 12;
+const THUMBPRINT_SLEEP_INTERVAL: number = 5000; // In milliseconds.
+
+// Get the URL's issuing CA cert thumbprint.
+// This is used for OIDC provider configuration.
+async function getIssuerCAThumbprint(issuerUrl: pulumi.Output<string>, retriesLeft: number, interval: number): Promise<string> {
```

Comment:
This should probably return an Output.  In particular, pretty sure we do want dependency and secret information to flow through this function - so keeping everything in `Output` will be cleaner.

This should probably also throw if it fails instead of returning `""` - that would clean up the consuming code as well.

Reply:
Both changes are done. Thanks for the suggestion.